cmd/root_test.go

```diff
@@ -537,7 +532,7 @@ func TestBustCacheOnFormatterChange(t *testing.T) {
 	assertStats(t, as, statz, 32, 0, 0, 0)
 }
 
-func TestGitWorktree(t *testing.T) {
+func TestGit(t *testing.T) {
```

Comment:
If I'm reading this test correctly, it doesn't test out the situation where we have a single file with some staged changes *and* some unstaged changes. In particular, I think the scenario where the unstaged and staged changes conflict (post formatting) is most interesting.

I haven't read the implementation yet (I'm reading the tests first), so maybe this isn't actually interesting, but as a user, it *seems* interesting. Concretely, I'm thinking of something like this:

1. Create a new `test.py` file in need of some formatting:

   ```python
   print(   "woah there are a lot of spces there")
   ```

2. `git add test.py`: stage that file
3. Now edit `test.py` to fix the typo:

   ```python
   print(   "woah there are a lot of spaces there")
   ```
4. Do *not* stage that change.
5. Now invoke treefmt with `--walk=git_staged`. Does this work? It feels distressing to me because the unstaged "diff" wouldn't apply cleanly on top of the formatted staged changed. But perhaps I'm probably incorrectly assuming this stuff works on diffs. I need to go read your code now.